### PR TITLE
re-export DA interfaces from search-core

### PR DESCRIPTION
**Pull request by yen-tt**
- update search-core version to be 2.0.0-alpha.215 which the new DA interfaces. re-export those DA interfaces in headless.
- fix some new TS issues in DA jest test and existing issues in facets and verticalSearch test.

bump headless version to 2.0.0-alpha.131, to be publish after this pr is merged

J=SLAP-2324
TEST=none

### CODE REVIEW

File: tests/integration/facets.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ it('can select a facet option', () => {
   const [initialState, facetOption] = createInitialState(false);
   const answers = createMockedHeadless({}, initialState);
   answers.setFacetOption('testFieldId', facetOption, true);
-  expect(answers.state.filters.facets[0].options[0]).toEqual({
+  expect(answers.state.filters.facets?.[0].options[0]).toEqual({
```
Comment (oshi97) [inline tests/integration/facets.ts:10]: Out of curiosity did these TS issues come from a previous commit?

**Review by oshi97 — APPROVED** [review]

File: tests/integration/facets.ts (hunk 1 above)
Comment (yen-tt) [reply tests/integration/facets.ts:10]: yup, I can see these TS issues when I switch to develop branch just now.

File: etc/search-headless.api.md [hunk 2]
```diff
@@ -11,13 +11,29 @@ export interface AdditionalHttpHeaders {
     'Client-SDK'?: ClientSDKHeaderValues;
 }
 
+// Warning: (ae-forgotten-export) The symbol "Address" needs to be exported by the entry point index.d.ts
```
Comment (nmanu1) [inline etc/search-headless.api.md:14]: looks like there are some missing exports

File: etc/search-headless.api.md [hunk 3]
```diff
@@ -11,13 +11,29 @@ export interface AdditionalHttpHeaders {
     'Client-SDK'?: ClientSDKHeaderValues;
 }
 
+// Warning: (ae-forgotten-export) The symbol "Address" needs to be exported by the entry point index.d.ts
+//
+// @public
+export interface AddressDirectAnswer extends BaseFieldValueDirectAnswer<Address> {
+    // Warning: (ae-forgotten-export) The symbol "EnumOrLiteral" needs to be exported by the entry point index.d.ts
```
Comment (nmanu1) [inline etc/search-headless.api.md:18]: if we export `EnumOrLiteral` from Core, I think we can remove the duplicate model defined in Headless

Comment (yen-tt) [reply etc/search-headless.api.md:18]: addressed in another pr

File: etc/search-headless.api.md (hunk 2 above)
Comment (yen-tt) [reply etc/search-headless.api.md:14]: addressed in another pr

**Pull request merged**